zerver/openapi/zulip.yaml

```diff
@@ -448,6 +448,206 @@ paths:
+                      }
+    post:
+      operationId: create_drafts
+      tags: ["drafts"]
+      description: |
+        Create one of more drafts for the current user.
+
+        `POST {{ api_url }}/v1/drafts`
+      parameters:
+        - name: drafts
+          in: query
+          description: |
+            A json-encoded list of all of the drafts that should be created for the current user.
+          schema:
+            type: array
+            items:
+              $ref: "#/components/schemas/Draft"
+      responses:
+        "200":
+          description: Success.
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: "#/components/schemas/JsonSuccess"
+                  - properties:
+                      ids:
+                        type: array
+                        description: |
+                          An array of the IDS for the drafts that were just created in the same
+                          order as they were requested.
+                        items:
+                          type: integer
+                  - example: {"result": "success", "msg": "", "ids": [1, 2, 3]}
+        "400":
+          description: Bad request.
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: "#/components/schemas/CodedError"
+                  - example:
+                      {
+                        "code": "BAD_REQUEST",
+                        "msg": "Must specify exactly 1 stream ID for stream messages",
+                        "result": "error",
+                      }
+  /drafts/{draft_id}:
+    patch:
+      operationId: edit_draft
+      tags: ["drafts"]
+      description: |
+        Edit a single draft.
+
+        `PATCH {{ api_url }}/v1/drafts/{draft_id}`
+      parameters:
+        - name: draft_id
+          in: path
+          schema:
+            type: integer
+          description: |
+            The ID of the draft you want to edit.
+          required: True
+        - name: draft
+          in: query
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Draft"
+          description: |
+            A JSON-encoded string denoting the edited data you
+            want to save for the given draft ID.
+          required: True
+      responses:
+        "200":
+          description: Success.
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: "#/components/schemas/JsonSuccess"
+        "404":
+          description: Not Found.
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - properties:
+                      result:
+                        type: string
+                        description: |
+                          Literally just the word "error".
```

Comment:
I think it'd be better to use the standard JsonError schema rather than documenting what specific string we'll use.

Reply:
Also in general we should be using [`enum`](https://swagger.io/docs/specification/data-models/enums/) (as `JsonError` does) to document when a string has a specific value or a small set of specific values.